lib/db/sstable_files.cc

```diff
@@ -0,0 +1,76 @@
+#include "db/sstable_files.hh"
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <fcntl.h>
+#include <iostream>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <tuple>
+#include <unistd.h>
+
+using namespace dibibase::db;
+
+SSTableFiles::SSTableFiles(char *data, uint8_t sstable_num)
+    : m_data(data), m_sstable_num(sstable_num), m_dbname("dibibase") {
+
+  for (uint8_t index = 0; index < 4; ++index) {
```

Comment:
I don't think you have to conscious about the integer size in for loops. It's gonna be on the stack anyways.
I'm not sure, but I think the compiler will optimize it anyways to the platforms word size `uint64_t` in our case.

Reply:
Is it better to use `uint64_t `instead of `uint8_t` ?

Reply:
You can use `uint_fast8_t`. Or just `long` and let the compiler do it's job. Or `auto`, but this defaults to unsigned types. Works here. May cause problems elsewhere.